api/envoy/type/v3/format.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package envoy.type.v3;
+
+import "google/protobuf/struct.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.type.v3";
+option java_outer_classname = "FormatProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Format]
+
+// Allows to define one of format: flat plain text or structured json.
+message StringOrJson {
+  oneof format {
+    // Allows to specify flat plain text format.
+    //
+    // .. code-block::
+    //
+    //   string_format: %RESP_BODY% %RESPONSE_CODE% %RESPONSE_FLAGS% My custom response
```

Comment:
Can you link to where the possible format placeholders are defined? I'm not completely sold that this should be a core type, mainly because it has implicit dependencies on the access logging system? I could go either way on this. It might better to move it closer to the access logging config? cc @htuch

Reply:
I think this could live in `api/envoy/config/core`. Or at least somewhere that makes it clear that (1) it's not context free (so not in `envoy.type`) and (2) that it is shared, lots of folks basically use this, header format, access log, now response mapper...

Reply:
Thanks. @htuch I will use a separate PR to move `access_log_format` config to `/api/envoy/config/core`

Reply:
+1 let's do that.